src/AbstractWalkers.jl

```diff
@@ -0,0 +1,147 @@
+"""
+Module containing abstract definitions for walkers.
+"""
+module AbstractWalkers
+
+using AtomsBase
+using Unitful
+using ..Potentials
+using ..EnergyEval
+
+export AtomWalker, AtomWalkers, LJAtomWalkers
+export update_walker!
+
+abstract type AtomWalkers end
+
+"""
+    mutable struct AtomWalker
+
+The `AtomWalker` struct represents a walker composed of atoms/molecules.
+
+# Fields
+- `configuration::FastSystem`: The configuration of the walker.
+- `energy::typeof(0.0u"eV")`: The energy of the walker.
+- `iter::Int64`: The current iteration number of the walker.
+- `num_frozen_part::Int64`: The number of frozen particles in the walker.
+- `energy_frozen_part::typeof(0.0u"eV")`: The energy of the frozen particles in the walker, serves as a constant energy offset
+    to the interacting part of the system.
+
+# Constructor
+```julia
+AtomWalker(configuration::FastSystem; energy=0.0u"eV", iter=0, num_frozen_part=0, energy_frozen_part=0.0u"eV")
+```
+Create a new `AtomWalker` with the given configuration and optional energy, iteration number, number of frozen particles, and energy of the frozen particles.
+
+"""
+mutable struct AtomWalker 
+    configuration::FastSystem
+    energy::typeof(0.0u"eV")
+    iter::Int64
+    num_frozen_part::Int64
+    energy_frozen_part::typeof(0.0u"eV")
+    function AtomWalker(configuration::FastSystem; energy=0.0u"eV", iter=0, num_frozen_part=0, energy_frozen_part=0.0u"eV")
+        return new(configuration, energy, iter, num_frozen_part, energy_frozen_part)
```

Comment:
Where is the function `new` defined?

Reply:
`new` is a Julia special function that creates the object that's being defined (an inner constructor). More info [here](https://docs.julialang.org/en/v1/manual/constructors/#:~:text=It%20has%20access%20to%20a%20special%20locally%20existent%20function%20called%20new%20that%20creates%20objects%20of%20the%20block%27s%20type.).